tests/Unit/ParallelAlgorithms/Events/Test_ObserveVolumeIntegrals.cpp

```diff
@@ -0,0 +1,261 @@
+    results.volume_integrals = std::get<2>(reduction_data.data());
+  }
+};
+
+// Need forward declaration to avoid linking error.
+MockContributeReductionData::Results MockContributeReductionData::results{};
+
+template <typename Metavariables>
+struct ElementComponent {
+  using component_being_mocked = void;
+
+  using metavariables = Metavariables;
+  using array_index = int;
+  using chare_type = ActionTesting::MockArrayChare;
+  using phase_dependent_action_list =
+      tmpl::list<Parallel::PhaseActions<typename Metavariables::Phase,
+                                        Metavariables::Phase::Initialization,
+                                        tmpl::list<>>>;
+};
+
+template <typename Metavariables>
+struct MockObserverComponent {
+  using component_being_mocked = observers::Observer<Metavariables>;
+  using replace_these_simple_actions =
+      tmpl::list<observers::Actions::ContributeReductionData>;
+  using with_these_simple_actions = tmpl::list<MockContributeReductionData>;
+
+  using metavariables = Metavariables;
+  using array_index = int;
+  using chare_type = ActionTesting::MockArrayChare;
+  using phase_dependent_action_list =
+      tmpl::list<Parallel::PhaseActions<typename Metavariables::Phase,
+                                        Metavariables::Phase::Initialization,
+                                        tmpl::list<>>>;
+};
+
+template <typename System>
+struct Metavariables {
+  using component_list = tmpl::list<ElementComponent<Metavariables>,
+                                    MockObserverComponent<Metavariables>>;
+  using const_global_cache_tags = tmpl::list<>;  //  unused
+  enum class Phase { Initialization, Testing, Exit };
+
+  struct ObservationType {};
+  using element_observation_type = ObservationType;
+};
+
+template <size_t Dim>
+struct SomeSystem {
+  struct ScalarVar : db::SimpleTag {
+    using type = Scalar<DataVector>;
+  };
+
+  struct VectorVar : db::SimpleTag {
+    using type = tnsr::I<DataVector, Dim>;
+  };
+
+  struct TensorVar : db::SimpleTag {
+    using type = tnsr::Ij<DataVector, Dim>;
+  };
+
+  static constexpr size_t volume_dim = Dim;
+  using vars_for_test = tmpl::list<ScalarVar, VectorVar, TensorVar>;
+};
+
+template <typename System, typename ObserveEvent>
+void test_observe(const std::unique_ptr<ObserveEvent> observe) noexcept {
+  constexpr size_t volume_dim = System::volume_dim;
+  using metavariables = Metavariables<System>;
+  using element_component = ElementComponent<metavariables>;
+  using observer_component = MockObserverComponent<metavariables>;
+  using vars_for_test = typename System::vars_for_test;
+
+  const typename element_component::array_index array_index(0);
+  const ElementId<volume_dim> element_id{array_index};
+
+  // ObserveVolumeIntegrals requires some domain items in the DataBox.
+  // Any domain, mesh and basis should be fine--we'll just check received data
+  const domain::creators::Brick brick({{-0.5, -0.5, -0.5}}, {{0.5, 0.5, 0.5}},
```

Comment:
I *think* (haven't check carefully) this is the only place that assumes 3d. What I've done to get tests in multiple dimensions working more easily is to factor the domain creation out as:
```cpp
template <size_t Dim>
Domain<Dim> create_domain();

template <>
Domain<1> create_domain() {
  return Interval{....}.create_domain();
}

template <>
Domain<2> create_domain() {
  return Rectangle{...}.create_domain();
}

template <>
Domain<3> create_domain() {
  return Brick{....}.create_domain();
}
```
Then you should be able to factor the `SPECTRE_TEST_CASE` into a function:
```cpp
template <size_t Dim>
void test() {
  .....
}
```
and in the `SPECTRE_TEST_CASE` do:
```cpp
  test<1>();
  test<2>();
  test<3>();
```
What do you think?

Reply:
I guess I can just a lambda that gives the right domain for the given dimension. The rest would be the same, as you noticed.

Reply:
Yea, that also works. I've found the function template specializations a bit easier to read than the `make_overloader` calls, but here it really doesn't make much difference so do whichever you prefer :)

Reply:
By the way, clang-tidy is complaining about `std::move`ing `mesh`, as I am also using it to compute the expected integrals and the order of both operations is apparently not guaranteed. Should I just not move `mesh` then?

Reply:
Could you elaborate on what exactly the complaint is?

Reply:
Nevermind. I removed all the clang-tidy issues by removing all the `std::move`. Should be good now.